fix: Block Manual Enrollment of learners in open edX

aliraza-abbasi opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [ ] Settings
  - [ ] New settings have reasonable development defaults, if applicable
  - [ ] Opened issue in the [ol-infrastructure repo](https://github.com/mitodl/ol-infrastructure/issues/new) for any necessary configuration changes to deployed environments

#### What are the relevant tickets?
https://github.com/mitodl/hq/issues/1095

#### What's this PR do?
Fixes that Instructors should not be allowed to enroll users manually in open edX through the instructor dashboard. For consistency, all enrollments should be done through the mitxonline application.

#### How should this be manually tested?
- Setup the theme
- Add course in the `studio` as a non-admin user
- Visit the instructor dashboard
- Validate the `Enrollment Section` should not be there
- Repeat the above steps with an admin user
- Validate the `Enrollment Section` is available for the admin user

#### Screenshots (if appropriate)
Before:
<img width="1546" alt="Screenshot 2023-03-31 at 1 07 27 PM" src="https://user-images.githubusercontent.com/77275478/229064246-f04aa234-5422-4b35-acfd-4efd799600f0.png">

After:
<img width="1532" alt="Screenshot 2023-04-04 at 4 39 28 PM" src="https://user-images.githubusercontent.com/77275478/229780543-640c680e-2546-4d6c-b8aa-e0cbade24168.png">

pdpinch commented:
Since we're editing the template, can we add a message about Batch Enrollment? Something like: "Batch Enrollment is disabled. Please ask users to enroll via mitxonline.mit.edu or contact mitx-et@mit.edu"

aliraza-abbasi commented:
> Since we're editing the template, can we add a message about Batch Enrollment? Something like: "Batch Enrollment is disabled. Please ask users to enroll via mitxonline.mit.edu or contact [mitx-et@mit.edu](mailto:mitx-et@mit.edu)"

@pdpinch I've added the message and updated the screenshot. please have a look.

pdpinch commented on lms/templates/instructor/instructor_dashboard_2/membership.html:
```diff
@@ -5,9 +5,16 @@
+    % if not section_data['access']['admin']:
+    <label>
+      ${_("Batch Enrollment is disabled. Please ask users to enroll via ")}
+      <a href="https://mitxonline.mit.edu">mitxonline.mit.edu</a>
```
Is there some way to pull this link from the environment so that it will work on RC and production? 

If it takes more than 5 minutes effort, it's not worth doing.

aliraza-abbasi replied:
> Is there some way to pull this link from the environment so that it will work on RC and production?
> 
> If it takes more than 5 minutes effort, it's not worth doing.

It was already in the environment. I just added it now. Sorry for missing it before

pdpinch commented on lms/templates/instructor/instructor_dashboard_2/membership.html:
```diff
@@ -0,0 +1,299 @@
+    % if not section_data['access']['admin']:
+    <label>
+      ${_("Batch Enrollment is disabled. Please ask users to enroll via ")}
+      <a href="${settings.MARKETING_SITE_BASE_URL}">MITxOnline</a>
```
```suggestion
      <a href="${settings.MARKETING_SITE_BASE_URL}">MITx Online</a>
```

pdpinch reviewed (approved):
After the small typo is fixed, I am fine with this. 

You should have someone do a code review before it's merged though, just to make sure it doesn't break the theme.

arslanashraf7 commented on lms/templates/instructor/instructor_dashboard_2/membership.html:
```diff
@@ -0,0 +1,299 @@
+      ${_("Batch Enrollment is disabled. Please ask users to enroll via ")}
+      <a href="${settings.MARKETING_SITE_BASE_URL}">MITx Online</a>
+      ${_(" or contact ")}
+      <a href="mailto:mitx-et@mit.edu">mitx-et@mit.edu</a>
```
A couple of thoughts I wanted to bring up:

- I haven't seen this address before, We've used https://github.com/mitodl/mitxonline/blob/main/frontend/public/src/components/ReceiptPageDetailCard.js#L59 in MITxOnline though.
- Thought this could be a configurable value as well. I see [in `ol-infrastructure`](https://github.com/mitodl/ol-infrastructure/blame/f3264070d7b197b15ada2484cd249148ff70a47c/src/bilder/images/edxapp/templates/edxapp/mitxonline/common_values.yml.tmpl#L406)  but this is very old. I think we can go with this hard-coded address if we're sure that it's not going to change any time sooner or more frequently.

FYI, @pdpinch

pdpinch replied:
> I think we can go with this hard-coded address if we're sure that it's not going to change any time sooner or more frequently.

I agree. 

mitx-et@mit.edu is a support address for course faculty.

arslanashraf7 reviewed (approved):
Looks good 👍 Just more of a question type comment left.
